pkg/kontext/dockerignore_test.go

```diff
@@ -0,0 +1,210 @@
+
+	patterns, err := ignorablePatterns(ignoreFile)
+
+	if err != nil {
+		t.Error("ignorablePatterns() = ", err)
+	}
+
+	if !reflect.DeepEqual(ignored, patterns) {
+		t.Errorf("Expecting ignore patterns %v but got %v", ignored, patterns)
+	}
+}
+
+func TestEmptyDockerIgnoredPatterns(t *testing.T) {
+
+	wd, err := os.Getwd()
+	if err != nil {
+		t.Fatal("os.Getwd() =", err)
+	}
+	ignoreFile := filepath.Join(wd, "ignoretestdata", "empty", ".dockerignore")
+
+	patterns, err := ignorablePatterns(ignoreFile)
+
+	if err != nil {
+		t.Error("ignorablePatterns() = ", err)
+	}
+
+	if len(patterns) != 0 {
+		t.Errorf("Expecting empty patterns but got %v", patterns)
+	}
+}
+
+func TestIgnoreables(t *testing.T) {
+	wd, err := os.Getwd()
+	if err != nil {
+		t.Fatal("os.Getwd() =", err)
+	}
+	dir := filepath.Join(wd, "ignoretestdata", "dir2")
+	patterns, err := ignorablePatterns(dir + "/.dockerignore")
+
+	var excludedDirs = &[]string{}
+
+	if err != nil {
+		t.Error("isIgnorable() = ", err)
+	}
+
+	bundlefile := &bundleFile{
+		name:         "lib",
+		path:         filepath.Join(dir, "lib"),
+		rootDir:      dir,
+		patterns:     patterns,
+		excludedDirs: excludedDirs,
+	}
+
+	b := bundlefile.isIgnorable()
+
+	if !b {
+		t.Errorf("%s has to be ignored but its not.", bundlefile.name)
+	}
+
+	*excludedDirs = append(*excludedDirs, "lib")
+
+	bundlefile.name = "tempA"
+	bundlefile.path = filepath.Join(dir, "tempA")
+	bundlefile.name = "tempA"
+
+	b = bundlefile.isIgnorable()
+
+	if !b {
+		t.Errorf("%s has to be ignored but its not.", bundlefile.name)
+	}
+
+	bundlefile.name = "tempABC"
+	bundlefile.path = filepath.Join(dir, "tempABC")
+	bundlefile.name = "tempABC"
+	*excludedDirs = append(*excludedDirs, "tempA")
+
+	b = bundlefile.isIgnorable()
+
+	if b {
+		t.Errorf("%s should not be ignored but actually it is.", bundlefile.name)
+	}
+
+	bundlefile.name = "one.md"
+	bundlefile.path = filepath.Join(dir, "one.md")
+	bundlefile.name = "one.md"
+
+	b = bundlefile.isIgnorable()
+
+	if !b {
+		t.Errorf("%s has to be ignored but its not.", bundlefile.name)
+	}
+
+	bundlefile.name = "README.md"
+	bundlefile.path = filepath.Join(dir, "README.md")
+	bundlefile.name = "README.md"
+
+	b = bundlefile.isIgnorable()
+
+	if b {
+		t.Errorf("%s should not be ignored but actually it is.", bundlefile.name)
+	}
+
+	bundlefile.name = "target"
+	bundlefile.path = filepath.Join(dir, "target")
+	bundlefile.name = "target"
+
+	b = bundlefile.isIgnorable()
+
+	if !b {
+		t.Errorf("%s has to be ignored but its not.", bundlefile.name)
+	}
+
+	bundlefile.name = "target/foo-runner.jar"
+	bundlefile.path = filepath.Join(dir, "target", "foo-runner.jar")
+	bundlefile.name = "foo-runner.jar"
+	*excludedDirs = append(*excludedDirs, "target")
+
+	b = bundlefile.isIgnorable()
+
+	if b {
+		t.Errorf("%s should not be ignored but actually it is.", bundlefile.name)
+	}
+
+}
+
+func difference(slice1, slice2 []string) []string {
```

Comment:
Doesn't look like order matters, so maybe use `sets.NewString` from `k8s.io/apimachinery` which has a difference method?

Reply:
this is a cool pointer, I was looking for some collections that already has it , glad we have it in k8s :D